Error when install: Error: Build error: Failed compiling object src/term.o

Here is the console log. I wonder why it failed.
I'm using "x86_64-8.1.0-release-posix-seh-rt_v6-rev0.7z" "lua-5.4.2_Win64_vc17_lib.zip" and "lua-5.4.2_Win64_bin.zip".
Would anyone be able to help?
``` console
luarocks install luasystem
Installing https://luarocks.org/luasystem-0.4.1-1.src.rock

luasystem 0.4.1-1 depends on lua >= 5.1 (5.4-1 provided by VM: success)
x86_64-w64-mingw32-gcc -O2 -c -o src/core.o -ID:\Program Files\Lua/include src/core.c -DWINVER=0x0600 -D_WIN32_WINNT=0x0600 -Ic:\windows\system32\include -Ic:\windows\system32\include
x86_64-w64-mingw32-gcc -O2 -c -o src/compat.o -ID:\Program Files\Lua/include src/compat.c -DWINVER=0x0600 -D_WIN32_WINNT=0x0600 -Ic:\windows\system32\include -Ic:\windows\system32\include
x86_64-w64-mingw32-gcc -O2 -c -o src/time.o -ID:\Program Files\Lua/include src/time.c -DWINVER=0x0600 -D_WIN32_WINNT=0x0600 -Ic:\windows\system32\include -Ic:\windows\system32\include
x86_64-w64-mingw32-gcc -O2 -c -o src/environment.o -ID:\Program Files\Lua/include src/environment.c -DWINVER=0x0600 -D_WIN32_WINNT=0x0600 -Ic:\windows\system32\include -Ic:\windows\system32\include
x86_64-w64-mingw32-gcc -O2 -c -o src/random.o -ID:\Program Files\Lua/include src/random.c -DWINVER=0x0600 -D_WIN32_WINNT=0x0600 -Ic:\windows\system32\include -Ic:\windows\system32\include
x86_64-w64-mingw32-gcc -O2 -c -o src/term.o -ID:\Program Files\Lua/include src/term.c -DWINVER=0x0600 -D_WIN32_WINNT=0x0600 -Ic:\windows\system32\include -Ic:\windows\system32\include
src/term.c:126:59: error: 'ENABLE_VIRTUAL_TERMINAL_INPUT' undeclared here (not in a function); did you mean 'ENABLE_LINE_INPUT'?
     {"CIF_VIRTUAL_TERMINAL_INPUT", CHECK_WIN_FLAG_OR_ZERO(ENABLE_VIRTUAL_TERMINAL_INPUT)},
                                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/term.c:109:38: note: in definition of macro 'CHECK_WIN_FLAG_OR_ZERO'
 #define CHECK_WIN_FLAG_OR_ZERO(flag) flag
                                      ^~~~
src/term.c:136:64: error: 'ENABLE_VIRTUAL_TERMINAL_PROCESSING' undeclared here (not in a function)
     {"COF_VIRTUAL_TERMINAL_PROCESSING", CHECK_WIN_FLAG_OR_ZERO(ENABLE_VIRTUAL_TERMINAL_PROCESSING)},
                                                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/term.c:109:38: note: in definition of macro 'CHECK_WIN_FLAG_OR_ZERO'
 #define CHECK_WIN_FLAG_OR_ZERO(flag) flag
                                      ^~~~
src/term.c:136:64: warning: initialization of 'long unsigned int' from 'const struct ls_RegConst *' makes integer from pointer without a cast [-Wint-conversion]
     {"COF_VIRTUAL_TERMINAL_PROCESSING", CHECK_WIN_FLAG_OR_ZERO(ENABLE_VIRTUAL_TERMINAL_PROCESSING)},
                                                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/term.c:109:38: note: in definition of macro 'CHECK_WIN_FLAG_OR_ZERO'
 #define CHECK_WIN_FLAG_OR_ZERO(flag) flag
                                      ^~~~
src/term.c:136:64: note: (near initialization for 'win_console_out_flags[2].value')
     {"COF_VIRTUAL_TERMINAL_PROCESSING", CHECK_WIN_FLAG_OR_ZERO(ENABLE_VIRTUAL_TERMINAL_PROCESSING)},
                                                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/term.c:109:38: note: in definition of macro 'CHECK_WIN_FLAG_OR_ZERO'
 #define CHECK_WIN_FLAG_OR_ZERO(flag) flag
                                      ^~~~
src/term.c:136:64: error: initializer element is not constant
     {"COF_VIRTUAL_TERMINAL_PROCESSING", CHECK_WIN_FLAG_OR_ZERO(ENABLE_VIRTUAL_TERMINAL_PROCESSING)},
                                                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/term.c:109:38: note: in definition of macro 'CHECK_WIN_FLAG_OR_ZERO'
 #define CHECK_WIN_FLAG_OR_ZERO(flag) flag
                                      ^~~~
src/term.c:136:64: note: (near initialization for 'win_console_out_flags[2].value')
     {"COF_VIRTUAL_TERMINAL_PROCESSING", CHECK_WIN_FLAG_OR_ZERO(ENABLE_VIRTUAL_TERMINAL_PROCESSING)},
                                                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/term.c:109:38: note: in definition of macro 'CHECK_WIN_FLAG_OR_ZERO'
 #define CHECK_WIN_FLAG_OR_ZERO(flag) flag
                                      ^~~~
src/term.c:137:64: error: 'DISABLE_NEWLINE_AUTO_RETURN' undeclared here (not in a function)
     {"COF_DISABLE_NEWLINE_AUTO_RETURN", CHECK_WIN_FLAG_OR_ZERO(DISABLE_NEWLINE_AUTO_RETURN)},
                                                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~
src/term.c:109:38: note: in definition of macro 'CHECK_WIN_FLAG_OR_ZERO'
 #define CHECK_WIN_FLAG_OR_ZERO(flag) flag
                                      ^~~~
src/term.c:137:64: warning: initialization of 'long unsigned int' from 'const struct ls_RegConst *' makes integer from pointer without a cast [-Wint-conversion]
     {"COF_DISABLE_NEWLINE_AUTO_RETURN", CHECK_WIN_FLAG_OR_ZERO(DISABLE_NEWLINE_AUTO_RETURN)},
                                                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~
src/term.c:109:38: note: in definition of macro 'CHECK_WIN_FLAG_OR_ZERO'
 #define CHECK_WIN_FLAG_OR_ZERO(flag) flag
                                      ^~~~
src/term.c:137:64: note: (near initialization for 'win_console_out_flags[3].value')
     {"COF_DISABLE_NEWLINE_AUTO_RETURN", CHECK_WIN_FLAG_OR_ZERO(DISABLE_NEWLINE_AUTO_RETURN)},
                                                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~
src/term.c:109:38: note: in definition of macro 'CHECK_WIN_FLAG_OR_ZERO'
 #define CHECK_WIN_FLAG_OR_ZERO(flag) flag
                                      ^~~~
src/term.c:137:64: error: initializer element is not constant
     {"COF_DISABLE_NEWLINE_AUTO_RETURN", CHECK_WIN_FLAG_OR_ZERO(DISABLE_NEWLINE_AUTO_RETURN)},
                                                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~
src/term.c:109:38: note: in definition of macro 'CHECK_WIN_FLAG_OR_ZERO'
 #define CHECK_WIN_FLAG_OR_ZERO(flag) flag
                                      ^~~~
src/term.c:137:64: note: (near initialization for 'win_console_out_flags[3].value')
     {"COF_DISABLE_NEWLINE_AUTO_RETURN", CHECK_WIN_FLAG_OR_ZERO(DISABLE_NEWLINE_AUTO_RETURN)},
                                                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~
src/term.c:109:38: note: in definition of macro 'CHECK_WIN_FLAG_OR_ZERO'
 #define CHECK_WIN_FLAG_OR_ZERO(flag) flag
                                      ^~~~
src/term.c:138:62: error: 'ENABLE_LVB_GRID_WORLDWIDE' undeclared here (not in a function)
     {"COF_ENABLE_LVB_GRID_WORLDWIDE", CHECK_WIN_FLAG_OR_ZERO(ENABLE_LVB_GRID_WORLDWIDE)},
                                                              ^~~~~~~~~~~~~~~~~~~~~~~~~
src/term.c:109:38: note: in definition of macro 'CHECK_WIN_FLAG_OR_ZERO'
 #define CHECK_WIN_FLAG_OR_ZERO(flag) flag
                                      ^~~~
src/term.c:138:62: warning: initialization of 'long unsigned int' from 'const struct ls_RegConst *' makes integer from pointer without a cast [-Wint-conversion]
     {"COF_ENABLE_LVB_GRID_WORLDWIDE", CHECK_WIN_FLAG_OR_ZERO(ENABLE_LVB_GRID_WORLDWIDE)},
                                                              ^~~~~~~~~~~~~~~~~~~~~~~~~
src/term.c:109:38: note: in definition of macro 'CHECK_WIN_FLAG_OR_ZERO'
 #define CHECK_WIN_FLAG_OR_ZERO(flag) flag
                                      ^~~~
src/term.c:138:62: note: (near initialization for 'win_console_out_flags[4].value')
     {"COF_ENABLE_LVB_GRID_WORLDWIDE", CHECK_WIN_FLAG_OR_ZERO(ENABLE_LVB_GRID_WORLDWIDE)},
                                                              ^~~~~~~~~~~~~~~~~~~~~~~~~
src/term.c:109:38: note: in definition of macro 'CHECK_WIN_FLAG_OR_ZERO'
 #define CHECK_WIN_FLAG_OR_ZERO(flag) flag
                                      ^~~~
src/term.c:138:62: error: initializer element is not constant
     {"COF_ENABLE_LVB_GRID_WORLDWIDE", CHECK_WIN_FLAG_OR_ZERO(ENABLE_LVB_GRID_WORLDWIDE)},
                                                              ^~~~~~~~~~~~~~~~~~~~~~~~~
src/term.c:109:38: note: in definition of macro 'CHECK_WIN_FLAG_OR_ZERO'
 #define CHECK_WIN_FLAG_OR_ZERO(flag) flag
                                      ^~~~
src/term.c:138:62: note: (near initialization for 'win_console_out_flags[4].value')
     {"COF_ENABLE_LVB_GRID_WORLDWIDE", CHECK_WIN_FLAG_OR_ZERO(ENABLE_LVB_GRID_WORLDWIDE)},
                                                              ^~~~~~~~~~~~~~~~~~~~~~~~~
src/term.c:109:38: note: in definition of macro 'CHECK_WIN_FLAG_OR_ZERO'
 #define CHECK_WIN_FLAG_OR_ZERO(flag) flag
                                      ^~~~

Error: Build error: Failed compiling object src/term.o

```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -28,6 +28,10 @@ The scope of what is covered by the version number excludes:
 
 ## Version history
 
+### unreleased
+
+- Chore: add compiler error on Windows if Virtual Terminal Processing is unavailable.
+
 ### Version 0.4.2, released 25-Jun-2024
 
 - Fix: include additional headers for some MinGW installations

--- a/src/term.c
+++ b/src/term.c
@@ -107,6 +107,11 @@ typedef struct ls_RegConst {
 // This is needed because some flags are not defined on all platforms. So we
 // still export the constants, but they will be all 0, and hence not do anything.
 #ifdef _WIN32
+// check compatibility: Windows virtual terminal processing was added in 2015,
+// some older compiler suites don't have the proper headers.
+#ifndef ENABLE_VIRTUAL_TERMINAL_INPUT
+#error Virtual terminal macros are undefined (eg. ENABLE_VIRTUAL_TERMINAL_INPUT). Update the toolchain or revert to Luasystem < 0.4
+#endif
 #define CHECK_WIN_FLAG_OR_ZERO(flag) flag
 #define CHECK_NIX_FLAG_OR_ZERO(flag) 0
 #else

